libraries/bsg_manycore_memory_manager.h

```diff
@@ -28,7 +28,9 @@
 
 namespace awsbwhal {
         class MemoryManager {
-                std::mutex mMemManagerMutex;
+                #ifdef _MMAN_MUTEX_
+                  std::mutex mMemManagerMutex;
+                #endif
```

Comment:
is this a standard macro? Or is this something we need to define now?

Reply:
I don't think it is a standard macro. We will need to define it separately. Is library.mk a good spot for this?

Reply:
Is there a more basic switch? Like if newlib?

Reply:
This might conflict with something in newlib. I guess, we can be safer and say if BLACKPARROT or something like that. What do you think?

Reply:
I like the macro how it is.

Reply:
Not standard (and doesn't conflict with anything): good
Descriptive (does exactly what the name would imply): good
Does the job: excellent
What's the problem?

Reply:
Well, the original question was if this was a standard macro which you just confirmed it isn't. So, there is no problem.